src/cryptonote_basic/miner.cpp

```diff
@@ -763,45 +784,54 @@ namespace cryptonote
     return (uint8_t)( ceil( (other * 1.f / total * 1.f) * 100) );    
   }
   //-----------------------------------------------------------------------------------------------------    
-  bool miner::on_battery_power()
+  boost::logic::tribool miner::on_battery_power()
   {
     #ifdef _WIN32
 
       SYSTEM_POWER_STATUS power_status;
     	if ( GetSystemPowerStatus( &power_status ) != 0 )
     	{
-        return power_status.ACLineStatus != 1;
+        return boost::logic::tribool(power_status.ACLineStatus != 1);
     	}
 
     #elif defined(__linux__)
 
       // i've only tested on UBUNTU, these paths might be different on other systems
       // need to figure out a way to make this more flexible
-      const std::string POWER_SUPPLY_STATUS_PATH = "/sys/class/power_supply/ACAD/online";
-
-      if( !epee::file_io_utils::is_file_exist(POWER_SUPPLY_STATUS_PATH) )
+      std::string power_supply_path = "";
+      const std::string POWER_SUPPLY_STATUS_PATHS[] = 
       {
-        LOG_ERROR("'" << POWER_SUPPLY_STATUS_PATH << "' file does not exist, can't determine if on AC power");
-        return false;
+        "/sys/class/power_supply/ACAD/online",
```

Comment:
The documentation is not very helpful, there really isn't a specification to this. And if you don't find the path, the code will have to assume AC power, so crap. Should this decision be left up to the user? Disabling mining themselves when unplugging?